### PR TITLE
fix 2377

### LINKED ISSUE #2377: <type>OnCreateInput is missing `@relationship` fields that are present on <type>CreateInput
**Describe the bug**
When using `connectOrCreate`, it should be possible to create a node as you would normally using a `create` mutation, however it seems that this is not possible. In particular, in case of connections of a node to itself, it is not possible to recursively allow relationships to be created using `connectOrCreate`, whereas it is possible to do so using `create`

**Type definitions**
Given the following definitions

```
  interface ResourceEntity {
    id: ID! @id(autogenerate: false)
    name: String
    """Allowed resource types (enums) """
    type: ResourceType!
    """
    Globally tracked tags for this resource (enum)
    """
    tags: [Tag!]
    properties: [Property!]
  }

  type Resource implements ResourceEntity {
    id: ID! @id(autogenerate: false)
    name: String
    type: ResourceType!
    externalIds: [ID!]
    """
    Globally tracked tags for this resource
    """
    tags: [Tag!]
    properties: [Property!]
    """
    Resources encapsulating the given resource (e.g., a github org contains a repo)
    """
    containedBy: [Resource!]! @relationship(type: "CONTAINS", direction: IN)
    createdAt: DateTime! @timestamp(operations: [CREATE])
    updatedAt: DateTime! @timestamp(operations: [UPDATE])
  }
```

the inputs auto-generated are as follows:

```
input ResourceCreateInput {
  id: ID!
  name: String
  externalIds: [ID!]
  properties: [Property!]
  type: ResourceType!
  tags: [Tag!]
  containedBy: ResourceContainedByFieldInput
}
...
input ResourceOnCreateInput {
  id: ID!
  name: String
  externalIds: [ID!]
  properties: [Property!]
}
```

**Expected behavior**
I would expect `ResourceOnCreateInput` to be the same as  `ResourceCreateInput`. Instead, any attributes defined in the interface are not available on the `ResourceOnCreateInput`

* OS: macOS X, neo4j: 5.1
* Version: @neo4j/graphql@3.10.1
* Node.js version: v18.11.

### PR DESCRIPTION
**Pull request by MacondoExpress**
# Includes enums in onCreateInput types

# Issue

`Closes #2377

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [x] Documentation has been updated
- [x] TCK tests have been updated
- [x] Integration tests have been updated
- [x] Example applications have been updated
- [x] New files have copyright header
- [x] CLA (https://neo4j.com/developer/cla/) has been signed

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at
 https://neo4j-graphql-toolbox-stage-2407.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Review by tbwiss — DISMISSED** [review]
LGTM! Thanks @MacondoExpress!

File: packages/graphql/tests/schema/enum.test.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ describe("Enum", () => {
                 INACTIVE
                 PENDING
             }
-
+            
             type Movie {
```
Comment (mjfwebb) [inline packages/graphql/tests/schema/enum.test.ts:34]: Can you please remove this change?

**Review by mjfwebb — CHANGES_REQUESTED** [review]

File: packages/graphql/tests/schema/enum.test.ts (hunk 1 above)
Comment (MacondoExpress) [reply packages/graphql/tests/schema/enum.test.ts:34]: Sure!

**Review by mjfwebb — APPROVED** [review]
Great, thanks!

**Pull request merged**